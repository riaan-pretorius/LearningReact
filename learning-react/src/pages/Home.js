import React, { Component } from "react";
import Jumbo from "../components/Jumbo";
import BodyItem from "../components/BodyItem";
import { Row, Col } from "reactstrap";

const bodyItems = [
  {
    heading: "Subheading",
    subText:
      "Donec id elit non mi porta gravida at eget metus. Maecenas faucibus mollis interdum."
  },
  {
    heading: "Subheading",
    subText:
      "Donec id elit non mi porta gravida at eget metus. Maecenas faucibus mollis interdum."
  },
  {
    heading: "Subheading",
    subText:
      "Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Cras mattis consectetur purus sit amet fermentum."
  },
  {
    heading: "Subheading",
    subText:
      "Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Cras mattis consectetur purus sit amet fermentum."
  },
  {
    heading: "Subheading",
    subText: "Maecenas sed diam eget risus varius blandit sit amet non magna."
  },
  {
    heading: "Subheading",
    subText: "Maecenas sed diam eget risus varius blandit sit amet non magna."
  }
];

class Home extends Component<{}, {}> {
  getBodyItems() {
    return bodyItems.map((item, index) => {
      return (
        <BodyItem key={index} heading={item.heading} subText={item.subText} />
      );
    });
  }
  render() {
    return (
      <Col>
        <Jumbo
          title="Jumbotron heading"
          subText="Cras justo odio, dapibus ac facilisis in, egestas eget quam. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus."
          buttonText="Sign up today"
        />
        <Row>{this.getBodyItems()}</Row>
      </Col>
    );
  }
}

export default Home;
