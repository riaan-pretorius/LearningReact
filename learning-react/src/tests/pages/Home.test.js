import React from 'react'
import Home from '../../pages/Home'
import {mount, shallow} from 'enzyme';
import renderer from 'react-test-renderer'

test('ContactUs Test Snapshot', () => {
  const component = renderer.create(
    <Home />
  )
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});

test('ContactUs Test', () => {
    const page = mount(
        <Home />
    )
  
    expect(page.find('.jumbotron').exists()).toBeTruthy()
  
    let header = page.find('h1')
    expect(header.exists()).toBeTruthy()
    expect(header.text()).toBe("Jumbotron heading")

    let button = page.find('button')
    expect(button.exists()).toBeTruthy()
    expect(button.text()).toBe("Sign up today")
})
