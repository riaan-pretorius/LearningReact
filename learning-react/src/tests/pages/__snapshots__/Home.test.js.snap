// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs Test Snapshot 1`] = `
<div
  className="col"
>
  <div
    className="jumbotron"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h1>
      Jumbotron heading
    </h1>
    <p>
      Cras justo odio, dapibus ac facilisis in, egestas eget quam. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.
    </p>
    <button
      className="btn btn-success"
      onClick={[Function]}
      type={undefined}
    >
      Sign up today
    </button>
  </div>
  <div
    className="row"
  >
    <div
      className="col col-sm-6"
    >
      <h4>
        Subheading
      </h4>
      <p>
        Donec id elit non mi porta gravida at eget metus. Maecenas faucibus mollis interdum.
      </p>
    </div>
    <div
      className="col col-sm-6"
    >
      <h4>
        Subheading
      </h4>
      <p>
        Donec id elit non mi porta gravida at eget metus. Maecenas faucibus mollis interdum.
      </p>
    </div>
    <div
      className="col col-sm-6"
    >
      <h4>
        Subheading
      </h4>
      <p>
        Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Cras mattis consectetur purus sit amet fermentum.
      </p>
    </div>
    <div
      className="col col-sm-6"
    >
      <h4>
        Subheading
      </h4>
      <p>
        Morbi leo risus, porta ac consectetur ac, vestibulum at eros. Cras mattis consectetur purus sit amet fermentum.
      </p>
    </div>
    <div
      className="col col-sm-6"
    >
      <h4>
        Subheading
      </h4>
      <p>
        Maecenas sed diam eget risus varius blandit sit amet non magna.
      </p>
    </div>
    <div
      className="col col-sm-6"
    >
      <h4>
        Subheading
      </h4>
      <p>
        Maecenas sed diam eget risus varius blandit sit amet non magna.
      </p>
    </div>
  </div>
</div>
`;
